### PR TITLE
G3-192 species endpoint

### PR DESCRIPTION
**Pull request by francastell**

### CODE REVIEW

File: src/geneweaver/api/controller/species.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+router = APIRouter(prefix="/species", tags=["species"])
+
+
+@router.get("/")
+def get_species(
+    cursor: Optional[deps.Cursor] = Depends(deps.cursor),
+    taxonomy_id: Annotated[
+        Optional[int], Query(format="int64", minimum=0, maxiumum=9223372036854775807)
+    ] = None,
+    gene_id_type: Optional[GeneIdentifier] = None,
```
Comment (bergsalex) [inline src/geneweaver/api/controller/species.py:19]: I think `reference_gene_id_type` would be a better name. This really will just check on what the reference type is, not all available types.

File: src/geneweaver/api/controller/species.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+) -> dict:
+    """Get species."""
+    response = species_service.get_species(cursor, taxonomy_id, gene_id_type)
```
Comment (bergsalex) [inline src/geneweaver/api/controller/species.py:22]: Just a note: I don't love that the API will return an int for the reference_gene_id_type, but I think we can tackle that in another PR. The right solution would probably be to use pydantic for the return but I would want to validate the schemas before we add that in.

File: src/geneweaver/api/controller/species.py (hunk 1 above)
Comment (francastell) [reply src/geneweaver/api/controller/species.py]: Changed

File: src/geneweaver/api/controller/species.py (hunk 2 above)
Comment (francastell) [reply src/geneweaver/api/controller/species.py]: Changed

**Review by bergsalex — APPROVED** [review]

**Pull request merged**